cranelift/codegen/src/isa/x86/abi.rs

```diff
@@ -519,11 +567,22 @@ fn fastcall_prologue_epilogue(func: &mut ir::Function, isa: &dyn TargetIsa) -> C
     // will adjust the stack pointer to make room for the rest of the required
     // space for this frame.
     let word_size = isa.pointer_bytes() as usize;
+    let num_fprs = csrs.iter(FPR).len();
     let csr_stack_size = ((csrs.iter(GPR).len() + 2) * word_size) as i32;
 
+    // Only create an FPR stack slot if we're going to save FPRs.
+    let fpr_slot = if num_fprs > 0 {
+        Some(func.create_stack_slot(ir::StackSlotData {
```

Comment:
@fitzgen and I had a [spirited conversation](https://github.com/bytecodealliance/cranelift/pull/1378#discussion_r382408324) about stack slots.

Unanswered question: what kind of stack slot should this be? `IncomingArg` has special meaning to `stack_layout` where it's assumed to exist only once. Contrary to other spill slot kinds, it doesn't get aligned to its size's next power of two, either. `IncomingArg` can be adjusted to be appropriate here, but using a different stack slot seemed to mix fewer concerns. Using a single stack slot would be additionally complicated here because Cranelift's notion of stack addresses is with respect to the function body, and would require additional bookkeeping in a prologue/epilogues.

Reply:
I've made the FPR stack slot an `ExplicitSlot` rather than `SpillSlot` because there _is_ some mention of `SpillSlot` being eligible for certain optimizations in `redundant_reload_remover`. There shouldn't be redundant reloads from this slot, but `ExplicitSlot` specifically indicates its use for `stack_load`/`stack_store`, which are the intended result of this PR (only not using those here because of encoding constraints that need to be fixed)

Reply:
(Reopening) Can you add a comment in the code expliciting your conclusion for choosing ExplicitSlot, please?

Reply:
@bnjbvr is the new comment sufficient?

Reply:
Yes, looks good, thanks!